k4Reco/Overlay/components/OverlayTiming.h

```diff
@@ -0,0 +1,157 @@
+      : MultiTransformer(
+            name, svcLoc,
+            {KeyValues("EventHeader", {"EventHeader"}), KeyValues("MCParticles", {"DefaultMCParticles"}),
+             KeyValues("SimTrackerHits", {"DefaultSimTrackerHits"}),
+             KeyValues("SimCalorimeterHits", {"DefaultSimCalorimeterHits"})},
+            {KeyValues("OutputMCParticles", {"NewMCParticles"}), KeyValues("OutputSimTrackerHits", {"MCParticles1"}),
+             KeyValues("OutputSimCalorimeterHits", {"MCParticles2"}),
+             KeyValues("OutputCaloHitContributions", {"OverlayCaloHitContributions"})}) {}
+
+  template <typename T> void overlayCollection(std::string collName, const podio::CollectionBase& inColl);
+
+  virtual StatusCode initialize() final;
+
+  retType virtual operator()(
+      const edm4hep::EventHeaderCollection& headers, const edm4hep::MCParticleCollection& mcParticles,
+      const std::vector<const edm4hep::SimTrackerHitCollection*>&     simTrackerHits,
+      const std::vector<const edm4hep::SimCalorimeterHitCollection*>& simCalorimeterHits) const final;
+
+  std::pair<float, float> define_time_windows(const std::string& Collection_name) const;
+
+private:
+  // These correspond to the index position in the argument list
+  constexpr static int TRACKERHIT_INDEX_POSITION = 2;
+  constexpr static int SIMCALOHIT_INDEX_POSITION = 3;
+
+  Gaudi::Property<bool>        m_randomBX{this, "RandomBx", false,
+                                   "Place the physics event at an random position in the train: overrides PhysicsBX"};
+  mutable Gaudi::Property<int> m_physBX{this, "PhysicsBX", 1, "Number of the Bunch crossing of the physics event"};
+  Gaudi::Property<int>         _nBunchTrain{this, "NBunchtrain", 1, "Number of bunches in a bunch train"};
+  // Gaudi::Property<int>         m_startWithBackgroundFile{this, "StartBackgroundFileIndex", -1,
+  //                                                "Which background file to startWith"};
+  Gaudi::Property<int> m_startWithBackgroundEvent{this, "StartBackgroundEventIndex", -1,
+                                                  "Which background event to startWith"};
+
+  Gaudi::Property<std::vector<std::vector<std::string>>> m_inputFileNames{
+      this, "BackgroundFileNames", {}, "Name of the edm4hep input file(s) with background."};
+
+  Gaudi::Property<std::vector<double>> m_Noverlay{
+      this, "NumberBackground", {}, "Number of Background events to overlay - either fixed or Poisson mean"};
+
+  Gaudi::Property<std::vector<bool>> m_Poisson{
+      this,
+      "Poisson_random_NOverlay",
+      {},
+      "Draw random number of Events to overlay from Poisson distribution with mean value NumberBackground"};
+
+  Gaudi::Property<std::string> _mcParticleCollectionName{
+      this, "BackgroundMCParticleCollectionName", "MCParticle",
+      "The name of the MCParticle collection in the background files"};
+
+  Gaudi::Property<float> _DefaultStart_int{
```

Comment:
Not used?

Reply:
It could be used, but since the time windows are passed for the collections we want to filter in time I think it's simpler just having one parameter for the time (the time window) instead of having the time window plus a default lower bound. I'm going to remove it for now; this is different from the original in iLCSoft.